File: backend/java/Vehicle_Configurator/src/main/java/com/example/demo/entities/VehicleModel.java
package com.example.demo.entities;

import jakarta.persistence.*;

@Entity
@Table(name = "vehicle_model")
public class VehicleModel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "model_id")
    private int modelId;

    @ManyToOne
    @JoinColumn(name = "manufacturer_id")
    private Manufacturer manufacturer;

    @ManyToOne
    @JoinColumn(name = "segment_id")
    private Segment segment;

    @Column(name = "model_name", nullable = false, length = 100)
    private String modelName;

    @Column(name = "image", nullable = false, length = 100)
    private String image;

    @Column(name = "base_price", nullable = false)
    private float basePrice;

    @Column(name = "minimum_quantity", nullable = false)
    private int minimumQuantity;

    public VehicleModel() {
    }

    public VehicleModel(int modelId, Manufacturer manufacturer, Segment segment, String modelName,
            String image, float basePrice, int minimumQuantity) {
        this.modelId = modelId;
        this.manufacturer = manufacturer;
        this.segment = segment;
        this.modelName = modelName;
        this.image = image;
        this.basePrice = basePrice;
        this.minimumQuantity = minimumQuantity;
    }

    public int getModelId() {
        return modelId;
    }

    public void setModelId(int modelId) {
        this.modelId = modelId;
    }

    public Manufacturer getManufacturer() {
        return manufacturer;
    }

    public void setManufacturer(Manufacturer manufacturer) {
        this.manufacturer = manufacturer;
    }

    public Segment getSegment() {
        return segment;
    }

    public void setSegment(Segment segment) {
        this.segment = segment;
    }

    public String getModelName() {
        return modelName;
    }

    public void setModelName(String modelName) {
        this.modelName = modelName;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public float getBasePrice() {
        return basePrice;
    }

    public void setBasePrice(float basePrice) {
        this.basePrice = basePrice;
    }

    public int getMinimumQuantity() {
        return minimumQuantity;
    }

    public void setMinimumQuantity(int minimumQuantity) {
        this.minimumQuantity = minimumQuantity;
    }

}

File: backend-dotnet/Models/VehicleModel.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend_dotnet.Models
{
    [Table("vehicle_model")]
    public class VehicleModel
    {
        [Key]
        [Column("model_id")]
        public int ModelId { get; set; }

        [Column("manufacturer_id")]
        public int ManufacturerId { get; set; }

        [ForeignKey(nameof(ManufacturerId))]
        public Manufacturer? Manufacturer { get; set; }

        [Column("segment_id")]
        public int SegmentId { get; set; }

        [ForeignKey(nameof(SegmentId))]
        public Segment? Segment { get; set; }

        [Required]
        [Column("model_name")]
        [MaxLength(100)]
        public string ModelName { get; set; } = string.Empty;

        [Required]
        [Column("image")]
        [MaxLength(100)]
        public string Image { get; set; } = string.Empty;

        [Column("base_price")]
        public float BasePrice { get; set; }

        [Column("minimum_quantity")]
        public int MinimumQuantity { get; set; }

        // Navigation Properties
        public ICollection<VehicleDetail> VehicleDetails { get; set; } = new List<VehicleDetail>();

        public ICollection<AlternateComponent> AlternateComponents { get; set; } = new List<AlternateComponent>();
    }
}
